scipy/linalg/tests/test_basic.py

```diff
@@ -628,7 +628,11 @@ def test_random_sym(self):
                 a[i, j] = a[j, i]
         for i in range(4):
             b = random([n])
-            x = solve(a, b, sym_pos=1)
+            with pytest.warns(
+                DeprecationWarning,
+                match="This key is deprecated and ``assume_a = 'pos'``",
+            ):
+                x = solve(a, b, sym_pos=1)
```

Comment:
What's keeping us from simply doing:
```suggestion
            x = solve(a, b, assume_a='pos')
```

Reply:
Surely we want both? Test the new behaviour and test that warning is being given

Reply:
Fair point. I'd suggest to adapt the existing tests to use the non-deprecated syntax, and add a test that specifically tests for the warning.